devices/src/legacy/serial.rs

```diff
@@ -181,26 +181,11 @@ impl Serial {
         }
         Ok(())
     }
-}
-
-impl RawIOHandler for Serial {
-    fn raw_input(&mut self, data: &[u8]) -> io::Result<()> {
-        if !self.is_loop() {
-            self.in_buffer.extend(data);
-            self.recv_data()?;
-        }
-        Ok(())
-    }
-}
-
-impl BusDevice for Serial {
-    fn read(&mut self, offset: u64, data: &mut [u8]) {
-        if data.len() != 1 {
-            METRICS.uart.missed_read_count.inc();
-            return;
-        }
 
-        data[0] = match (offset as u64) as u8 {
+    // Handles the read of one byte from the serial console.
```

Comment:
I would say it handles reading the register at the specified offset which in this case is 1 byte long.

Reply:
Yap, you're right, bad wording. Will update once I get more comments.